--- a/org/eclipse/jdt/internal/compiler/problem/ProblemReporter.java
+++ b/org/eclipse/jdt/internal/compiler/problem/ProblemReporter.java
@@ -2007,6 +2007,9 @@
 		ReferenceContext contextSave = this.referenceContext;
 		try {
 			for (AnnotationBinding annotationBinding : annotations.get()) {
+				if (annotationBinding == null) {
+					continue;
+				}
\ No newline at end of file
 				if (annotationBinding.getAnnotationType().id == TypeIds.T_JavaLangDeprecated) {
 					for (ElementValuePair elementValuePair : annotationBinding.getElementValuePairs()) {
 						if (CharOperation.equals(elementValuePair.getName(), TypeConstants.SINCE) && elementValuePair.value instanceof StringConstant)
